test/verify/check-testlib

```diff
@@ -103,6 +103,23 @@ class TestRunTest(MachineCase):
         self.assertIn(b"1..0", out)
         self.assertNotIn(b"TestExample", out)
 
+    def testRetry(self):
+        env = os.environ.copy()
+        try:
+            env["TEST_FAILURES"] = "1"
+            del env["TEST_JOBS"]
+        except KeyError:
+            pass
+        process = subprocess.run([run_tests, "--test-dir", VERIFY_DIR, "TestExample.testFail", "TestExample.testSkip"],
+                                 env=env, capture_output=True)
+        stdout = process.stdout
+        self.assertRegex(stdout, b"\nnot ok 1 .*test\/verify\/check-example TestExample.testFail # RETRY 1 \(be robust against unstable tests\)\n")
```

Comment:
strictly speaking, this should not work because this PR affects `test/verify/check-example` and therefore it should not be retried. I have fix for this. The only way we can get this in is by sending changes into `test/verify/check-example` as separate PR and then implementing these checks. I'll do that.

Reply:
oh man, I can now send PR with those changes, then deliver this test and it will work. But if anyone in the future touches `test/verify/check-example` it won't work. I have one more idea, /ME goes to try it out.

Reply:
oh no, I would have to do `git commit` on ones machine, which I really don't want to do. We actually have `git` in VMs, so I am now thinking about copying `test/` into VM where I create simple git repo for it and then can test it properly without being afraid of breaking ones files or commits.

Reply:
ok, that would be massive task, it also needs bots, then we would need to create set of tests that don't need VMs...
Maybe we can just simply say to our run-tests to never consider `test-example` as affected file. We won't loose anything by doing that and we can then happily use it for testing. Then of course I need to change different file for testing of "changed files are retried more time" but that is not a problem, as if it was not changed in the PR, we will fake some change and if it was, then we don't even need to fake change (but we as well might to keep it simple).

This is not a Friday after lunch topic it seems :)